src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -248,6 +259,8 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
                 //renew retention lease with global checkpoint so that any shard that picks up shard replication task has data until then.
                 try {
                     retentionLeaseHelper.renewRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint, followerShardId)
+                    followerClusterStats.stats[followerShardId]!!.localCheckpoint = indexShard.lastSyncedGlobalCheckpoint
+
```

Comment:
Can we extract renew + stats update into a member method? This is so we don't miss if we introduce newer paths to renew.

Reply:
It is difficult to do that as we are updating stats at 4 different places in 3 different scenarios :

1. GetChanges success
2. GetChanges failure and want to know the reason as well. 
3. Retention lease success